### ISSUE TITLE
Unable to download SoundCloud from ApkMirror

### ISSUE THREAD

**Issue by archit9169** [j-hc/revanced-magisk-module#487]
Tried adding soundcloud but it did not work. The link in build log redirects to correct page, so looks like it's not fetching the apk from there.

Is this something related to sig.txt file that was present in repo?

config.toml:
[SoundCloud]
apkmirror-dlurl = "https://www.apkmirror.com/apk/soundcloud/soundcloud-soundcloud/"

Build logs:
2024-08-15 16:28:54 URL:https://www.apkmirror.com/uploads/?appcategory=soundcloud-soundcloud [313347] -> "-" [1]
[+] Choosing version '2024.08.09-release' for SoundCloud
[+] Downloading 'SoundCloud' from apkmirror
2024-08-15 16:28:54 URL:https://www.apkmirror.com/apk/google-inc/photos/ [428308] -> "-" [1]
2024-08-15 16:28:55 URL:https://www.apkmirror.com/apk/soundcloud/soundcloud-soundcloud/soundcloud-soundcloud-2024-08-09-release-release/ [273569] -> "-" [1]
[-] ERROR: Could not download 'SoundCloud' from apkmirror with version '2024.08.09-release', arch 'all', dpi 'nodpi'
Error: utils.sh [-] ERROR: Could not download 'SoundCloud' from apkmirror with version '2024.08.09-release', arch 'all', dpi 'nodpi'

**Comment by Othmane-ElAlami** [comment]
I'm pretty sure it's just the arch that need to be changed, you can try universal or arm64-v8a

**Comment by archit9169** [comment]
> I'm pretty sure it's just the arch that need to be changed, you can try universal or arm64-v8a

As per the config file doc, the default is 'all' which is for universal arch. Also, if you follow the link, there is no other arch for soundcloud. 

Anyother insight is appreciated.

**Comment by Othmane-ElAlami** [comment]
> There is no other arch for SoundCloud

![image](https://github.com/user-attachments/assets/84fac724-f2dd-45f2-bd89-38704d0af7c3)

Just try, you haven't much to lose.

**Comment by archit9169** [comment]
> > There is no other arch for SoundCloud
> 
> ![image](https://private-user-images.githubusercontent.com/111425505/362319218-182ab7be-ed24-4d75-9992-73bd8263ed20.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4NjMwOTIsIm5iZiI6MTcyNDg2Mjc5MiwicGF0aCI6Ii8xMTE0MjU1MDUvMzYyMzE5MjE4LTE4MmFiN2JlLWVkMjQtNGQ3NS05OTkyLTczYmQ4MjYzZWQyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyOFQxNjMzMTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kMzFiNTUwMjIzYTM3MjU5YThmODM4OGRjMTBlMGQxOTllMGE3OGVhYjcwZTA3NzdkMmY1NGE3NWI4YzcwNzJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Pg4wAAewjYFLCQGWxytHEf8nvFh9hAMziD26OU4Q2JA)
> 
> Just give it a try; you don't have much to lose.

![Screenshot_20240828_220302_Brave](https://github.com/user-attachments/assets/67be78ae-4774-4b36-87ad-2c887a659056)

This is the latest release available on apkmirror, its universal nodpi bundeled apk. Same as before. What you posted were either older versions or beta versions which I do not want to download.

Chaning arch in config should not be converend with, as it displays universal on apkmirror so the default all option should work like in youtube case.

This might be just an issue with bundled apks or something else.

**Comment by j-hc** [comment]
supports bundles now 1792aa3

**Comment by E85Addict** [comment]
> supports bundles now [1792aa3](https://github.com/j-hc/revanced-magisk-module/commit/1792aa367b1cead30265a57248e7c86b6fe573a4)

I gave it a shot with Infinity for reddit, since recently they only provide bundles and it seems to fail.

Config.toml:

> [Infinity]
> apkmirror-dlurl = "https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/"


Build Log:

> 2024-08-31 12:39:26 URL:https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/ [268636] -> "-" [1]
> 2024-08-31 12:39:27 URL:https://www.apkmirror.com/uploads/?appcategory=infinity-for-reddit [317818] -> "-" [1]
> [+] Choosing version '7.3.4' for Infinity
> [+] Downloading 'Infinity' from apkmirror
> 2024-08-31 12:39:28 URL:https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/infinity-for-reddit-7-3-4-release/ [269212] -> "-" [1]
> [-] ERROR: Could not download 'Infinity' from apkmirror with version '7.3.4', arch 'all', dpi 'nodpi'
> ::error::utils.sh [-] ERROR: Could not download 'Infinity' from apkmirror with version '7.3.4', arch 'all', dpi 'nodpi'
> 
> [-] ABORT: All builds failed.
> ::error::utils.sh [-] ABORT: All builds failed.

**Comment by j-hc** [comment]
> > supports bundles now [1792aa3](https://github.com/j-hc/revanced-magisk-module/commit/1792aa367b1cead30265a57248e7c86b6fe573a4)
> 
> I gave it a shot with Infinity for reddit, since recently they only provide bundles and it seems to fail.
> 
> Config.toml:
> 
> > [Infinity]
> > apkmirror-dlurl = "https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/"
> 
> Build Log:
> 
> > 2024-08-31 12:39:26 URL:https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/ [268636] -> "-" [1]
> > 2024-08-31 12:39:27 URL:https://www.apkmirror.com/uploads/?appcategory=infinity-for-reddit [317818] -> "-" [1]
> > [+] Choosing version '7.3.4' for Infinity
> > [+] Downloading 'Infinity' from apkmirror
> > 2024-08-31 12:39:28 URL:https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/infinity-for-reddit-7-3-4-release/ [269212] -> "-" [1]
> > [-] ERROR: Could not download 'Infinity' from apkmirror with version '7.3.4', arch 'all', dpi 'nodpi'
> > ::error::utils.sh [-] ERROR: Could not download 'Infinity' from apkmirror with version '7.3.4', arch 'all', dpi 'nodpi'
> > [-] ABORT: All builds failed.
> > ::error::utils.sh [-] ABORT: All builds failed.

you need to set the dpi

**Comment by E85Addict** [comment]
> > > supports bundles now [1792aa3](https://github.com/j-hc/revanced-magisk-module/commit/1792aa367b1cead30265a57248e7c86b6fe573a4)
> > 
> > 
> > I gave it a shot with Infinity for reddit, since recently they only provide bundles and it seems to fail.
> > Config.toml:
> > > [Infinity]
> > > apkmirror-dlurl = "[apkmirror.com/apk/docile-alligator/infinity-for-reddit](https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/)"
> > 
> > 
> > Build Log:
> > > 2024-08-31 12:39:26 URL:[apkmirror.com/apk/docile-alligator/infinity-for-reddit](https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/) [268636] -> "-" [1]
> > > 2024-08-31 12:39:27 URL:[apkmirror.com/uploads?appcategory=infinity-for-reddit](https://www.apkmirror.com/uploads/?appcategory=infinity-for-reddit) [317818] -> "-" [1]
> > > [+] Choosing version '7.3.4' for Infinity
> > > [+] Downloading 'Infinity' from apkmirror
> > > 2024-08-31 12:39:28 URL:[apkmirror.com/apk/docile-alligator/infinity-for-reddit/infinity-for-reddit-7-3-4-release](https://www.apkmirror.com/apk/docile-alligator/infinity-for-reddit/infinity-for-reddit-7-3-4-release/) [269212] -> "-" [1]
> > > [-] ERROR: Could not download 'Infinity' from apkmirror with version '7.3.4', arch 'all', dpi 'nodpi'
> > > ::error::utils.sh [-] ERROR: Could not download 'Infinity' from apkmirror with version '7.3.4', arch 'all', dpi 'nodpi'
> > > [-] ABORT: All builds failed.
> > > ::error::utils.sh [-] ABORT: All builds failed.
> 
> you need to set the dpi

That worked, completely went over my head, appreciate that! Thank you for all your amazing work!

**Comment by E85Addict** [comment]
Built a module using a bundle, and got this error

```
- Copying zip to temp directory
- Installing infinity-revanced-magisk-v7.3.4-all.zip
- Current boot slot: _b
- Device is system-as-root
Archive:  /data/user/0/ouz.j.mgul.pd/cache/flash/install.zip
  inflating: module.prop
**********************
 Infinity ReVanced 
 by j-hc 
**********************
*******************
 Powered by Magisk 
*******************
Archive:  /data/user/0/ouz.j.mgul.pd/cache/flash/install.zip
  inflating: customize.sh
- Extracting module files
Archive:  /data/user/0/ouz.j.mgul.pd/cache/flash/install.zip
  inflating: ml.docilealligator.infinityforreddit.apk
  inflating: customize.sh
   creating: bin/
   creating: bin/arm/
  inflating: bin/arm/cmpr
   creating: bin/x86/
  inflating: bin/x86/cmpr
   creating: bin/x64/
  inflating: bin/x64/cmpr
   creating: bin/arm64/
  inflating: bin/arm64/cmpr
  inflating: base.apk
  inflating: service.sh
  inflating: module.prop
  inflating: uninstall.sh
  inflating: config

* Updating ml.docilealligator.infinityforreddit to 7.3.4
ERROR: install-commit failed
Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed to collect certificates from /data/app/vmdl1641356249.tmp/ml.docilealligator.infinityforreddit.apk: Attempt to get length of null array]
! Installation failed
```

**Comment by j-hc** [comment]
> Built a module using a bundle, and got this error
> 
> ```
> - Copying zip to temp directory
> - Installing infinity-revanced-magisk-v7.3.4-all.zip
> - Current boot slot: _b
> - Device is system-as-root
> Archive:  /data/user/0/ouz.j.mgul.pd/cache/flash/install.zip
>   inflating: module.prop
> **********************
>  Infinity ReVanced 
>  by j-hc 
> **********************
> *******************
>  Powered by Magisk 
> *******************
> Archive:  /data/user/0/ouz.j.mgul.pd/cache/flash/install.zip
>   inflating: customize.sh
> - Extracting module files
> Archive:  /data/user/0/ouz.j.mgul.pd/cache/flash/install.zip
>   inflating: ml.docilealligator.infinityforreddit.apk
>   inflating: customize.sh
>    creating: bin/
>    creating: bin/arm/
>   inflating: bin/arm/cmpr
>    creating: bin/x86/
>   inflating: bin/x86/cmpr
>    creating: bin/x64/
>   inflating: bin/x64/cmpr
>    creating: bin/arm64/
>   inflating: bin/arm64/cmpr
>   inflating: base.apk
>   inflating: service.sh
>   inflating: module.prop
>   inflating: uninstall.sh
>   inflating: config
> 
> * Updating ml.docilealligator.infinityforreddit to 7.3.4
> ERROR: install-commit failed
> Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed to collect certificates from /data/app/vmdl1641356249.tmp/ml.docilealligator.infinityforreddit.apk: Attempt to get length of null array]
> ! Installation failed
> ```

fixed now a8ddca3
